codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageFactory.java

```diff
@@ -0,0 +1,253 @@
+        return new PreloadStringsCache(stringsCache, byteBufCache);
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} for the given {@code content} using {@link RedisMessageType#SIMPLE_STRING}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createSimpleString(String content) {
+        ObjectUtil.checkNotNull(content, "content");
+        RedisMessage cached = preloadSimpleString.stringCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(RedisMessageType.SIMPLE_STRING, content);
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} for the given {@code content} using {@link RedisMessageType#SIMPLE_STRING}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createSimpleString(ByteBuf content) {
+        ObjectUtil.checkNotNull(content, "content");
+        RedisMessage cached = preloadSimpleString.byteBufCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(RedisMessageType.SIMPLE_STRING,
+                                                                content.toString(CharsetUtil.UTF_8));
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} for the given {@code content} using {@link RedisMessageType#ERROR}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createError(String content) {
+        ObjectUtil.checkNotNull(content, "content");
+        RedisMessage cached = preloadErrors.stringCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(RedisMessageType.ERROR, content);
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} from {@code content} using {@link RedisMessageType#ERROR}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createError(ByteBuf content) {
+        ObjectUtil.checkNotNull(content, "content");
+        RedisMessage cached = preloadErrors.byteBufCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(RedisMessageType.ERROR,
+                                                                content.toString(CharsetUtil.UTF_8));
+    }
+
+    /**
+     * Creates a {@link IntegerRedisMessage} for the given {@code content}.
+     *
+     * @param value the message content.
+     * @return a new {@link IntegerRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createInteger(long value) {
+        return new IntegerRedisMessage(value);
```

Comment:
if the set of integers are really that small then a `switch` statement should be fine, if we need something larger/more automated then we can use `LongObjectHashMap`